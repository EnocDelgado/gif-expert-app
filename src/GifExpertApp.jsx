import { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Ichigo']);

    const handleAddCategory = ( newCategory ) => {

        // if there is a value in the array nothing is done
        if ( categories.includes( newCategory) ) return;

        // with spreadOperator we can add a new value
        setCategories([ ...categories, newCategory])
    }

    return (
        <>
            {/* Title */}
            <h1>GifExpertApp</h1>

            {/* input */}
            <AddCategory onNewValue={ handleAddCategory } /> 

            {/* Gif list */}
           
            {
                // Here is how we display our elements in the DOM
                categories.map( category => {
                    return (
                        <GifGrid 
                            key={ category } 
                            category={ category }
                        />
                    )
                })
            }
        </>
    )
}
