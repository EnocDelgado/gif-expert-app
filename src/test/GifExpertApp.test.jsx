import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../GifExpertApp"

describe('Test on GifExpertApp', () => {

    test('should do match with snapshot', () => {

        const { container } = render( <GifExpertApp />)
        expect( container ).toMatchSnapshot();
    })

})