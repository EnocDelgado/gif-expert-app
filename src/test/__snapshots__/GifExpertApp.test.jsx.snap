// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifExpertApp should do match with snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Ichigo
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
